src/mainwindow.cpp

```diff
@@ -109,6 +114,34 @@ void MainWindow::showTabAndTop() {
     getTopWidget()->show();
 }
 
+void MainWindow::updateTabButtons()
+{
+    auto tabBar = getTabBar();
+    QRect tabBarRect = getTabBar()->rect();
+    QRect newButtonTabRect = tabBar->tabRect(tabBar->count() - 1);
+
+    /* Decision is made at half way of the new button tab for smoothness */
+    newButtonTabRect.setWidth(newButtonTabRect.width() / 2);
+    bool newTabVisible = tabBarRect.contains(newButtonTabRect);
+    if (mp_ui->newTabSideButton->isHidden())
+        mp_ui->newTabSideButton->setVisible(!newTabVisible);
+    else
+        mp_ui->newTabSideButton->setHidden(newTabVisible);
+
+    QRect firstTabRect = tabBar->tabRect(0);
+    QRect lastRealTabRect = tabBar->tabRect(tabBar->count() - 2);
+
+    bool firstVisible = tabBarRect.contains(firstTabRect);
+    bool lastVisible = tabBarRect.contains(lastRealTabRect);
+    bool noOverFlow = firstVisible && lastVisible;
+
+    mp_ui->prevTabButton->setHidden(noOverFlow);
+    mp_ui->nextTabButton->setHidden(noOverFlow);
+
+    mp_ui->prevTabButton->setDisabled(firstVisible);
+    mp_ui->nextTabButton->setDisabled(lastVisible);
```

Comment:
This makes sense only if the stated purpose of the `prevTabButton` and `nextTabButton` is to scroll the tabbar (though with an undesirable but hard-to-eliminate side-effect of changing the active tab). However if the users perceive the function of those buttons as switching to an adjacent tab via GUI, then the fact that that operation is disabled before the first/last tab has been reached becomes controversial.